### PR TITLE
factor out AIE

### PR DESCRIPTION
**Pull request by makslevental**
The proximal purpose of this PR is to make testing the AIR -> Async pipeline easier. Otherwise, it's probably good to have things factored such that minimal examples/tests/etc. can be run without a dep on AIE.

### CODE REVIEW

File: mlir/include/air/Conversion/AIRtoAIEPasses.td [hunk 1]
```diff
@@ -0,0 +1,156 @@
+//===- AIRtoAIEPasses.td -------------------------------------------*- tablegen -*-===//
+//
+// Copyright (C) 2022, Xilinx Inc. All rights reserved.
+// Copyright (C) 2022, Advanced Micro Devices, Inc. All rights reserved.
```
Comment (keryell) [inline mlir/include/air/Conversion/AIRtoAIEPasses.td:4]: Happy new year!

File: mlir/include/air/Conversion/CMakeLists.txt [hunk 2]
```diff
@@ -6,4 +6,11 @@ set(LLVM_TARGET_DEFINITIONS Passes.td)
...
-add_mlir_doc(Passes AIRConversionPasses ./ -gen-pass-doc)
\ No newline at end of file
+add_mlir_doc(Passes AIRConversionPasses ./ -gen-pass-doc)
+
+if(BUILD_WITH_AIE)
+  set(LLVM_TARGET_DEFINITIONS AIRtoAIEPasses.td)
+  mlir_tablegen(AIRToAIEPass.h.inc -gen-pass-decls -name AIRtoAIEConversion)
+  add_public_tablegen_target(AIRtoAIEConversionPassIncGen)
+  add_mlir_doc(Passes AIRtoAIEConversionPasses ./ -gen-pass-doc)
+endif()
```
Comment (keryell) [inline mlir/include/air/Conversion/CMakeLists.txt:16]: Add an EOL.

File: mlir/include/air/Dialect/AIR/CMakeLists.txt [hunk 3]
```diff
@@ -31,3 +31,11 @@ mlir_tablegen(AIRTransformOps.cpp.inc -gen-op-defs)
...
 
 add_mlir_doc(AIRTransformOps AIRTransformOps ./ -gen-op-doc)
+
+if(BUILD_WITH_AIE)
+  set(LLVM_TARGET_DEFINITIONS AIRtoAIETransformOps.td)
+  mlir_tablegen(AIRtoAIETransformOps.h.inc -gen-op-decls)
+  mlir_tablegen(AIRtoAIETransformOps.cpp.inc -gen-op-defs)
+  add_public_tablegen_target(AIRtoAIETransformOpsIncGen)
+  add_mlir_doc(AIRtoAIETransformOps AIRtoAIETransformOps ./ -gen-op-doc)
+endif()
```
Comment (keryell) [inline mlir/include/air/Dialect/AIR/CMakeLists.txt:41]: Missing EOL.

File: mlir/include/air/Dialect/AIR/AIRtoAIETransformOps.td [hunk 4]
```diff
@@ -0,0 +1,36 @@
+//===- AIRtoAIETransformOps.td ----------------------------------*- tablegen -*-===//
+//
+// Copyright (C) 2022, Xilinx Inc. All rights reserved.
+// Copyright (C) 2022, Advanced Micro Devices, Inc. All rights reserved.
```
Comment (keryell) [inline mlir/include/air/Dialect/AIR/AIRtoAIETransformOps.td:4]: I guess this has to be solved up-stream?

Comment (makslevental) [reply mlir/include/air/Dialect/AIR/AIRtoAIETransformOps.td:4]: not sure what you mean - these are copy-pasta from adjacent tablegen - i'm not a lawyer but is it the thing that these need to be updated annually? if so i can update the adjacent ones as well

**Pull request closed without merging**